auth/src/swig/auth.i

```diff
@@ -1313,11 +1225,14 @@ static CppInstanceManager<Auth> g_auth_instances;
     get { return CredentialInternal; }
   }
 
+  internal FirebaseAuth authProxy;
+
   /// The currently signed-in @ref FirebaseUser, or null if there isn't one.
   public FirebaseUser User {
-    // Both FirebaseAuth.CurrentUser and AuthResult.User returns null if the
-    // user is invalid.
-    get { return (UserInternal != null && UserInternal.IsValid()) ? UserInternal : null; }
+    // Return the Auth's User, since there can currently only be one user per Auth.
+    get {
+      return authProxy != null ? authProxy.CurrentUser : null;
+    }
   }
```

Comment:
I am expecting `SignInResult.User` would have exactly the same issue. I think we should properly fix it until we actually remove this API.

Reply:
Agreed.  SignInResult already does this logic, of using the authProxy and returning the User based on it instead.